bokeh/models/widgets/inputs.py

```diff
@@ -263,3 +263,35 @@ class DatePicker(InputWidget):
     callback = Instance(Callback, help="""
     A callback to run in the browser whenever the current date value changes.
     """)
+
+class TextareaInput(InputWidget):
+    ''' Single-line input widget.
+
+    '''
+
+    value = String(default="", help="""
+    Initial or entered text value.
+    """)
+
+    callback = Instance(Callback, help="""
+    A callback to run in the browser whenever the user unfocuses the TextInput
+    widget by hitting Enter or clicking outside of the text box area.
+    """)
```

Comment:
This probably needs discussion. It's now possible to add callbacks to arbitrary properties such as `value` and I think that should be the primary mechanism going forward. We've discussed deprecating these "ad-hoc" callbacks, so I don't think they should be added to any new models. But doing that will require a slight refactor to move `callback` off the base class (another reason it's not needed here)